packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -1661,6 +1662,17 @@ abstract class ModalRoute<T> extends TransitionRoute<T> with LocalHistoryRoute<T
   final GlobalKey _subtreeKey = GlobalKey();
   final PageStorageBucket _storageBucket = PageStorageBucket();
 
+  final ValueNotifier<EdgeInsets> _clipDetailsNotifier = ValueNotifier<EdgeInsets>(EdgeInsets.zero);
+
+  EdgeInsets getNewClipDetails(Size topLayerSize) {
+    return EdgeInsets.zero;
+  }
+
+  void updateClipDetails(Size topLayerSize) {
```

Comment:
probably name it to didChangeBarrierSemanticsClip, and the input should probably be EdgeInsets

Reply:
Thanks! Done.